src/main/java/com/booksaw/betterTeams/message/Formatter.java

```diff
@@ -0,0 +1,280 @@
+package com.booksaw.betterTeams.message;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.booksaw.betterTeams.Main;
+
+import me.clip.placeholderapi.PlaceholderAPI;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import net.md_5.bungee.api.ChatColor;
+
+public class Formatter {
+
+    private static final Pattern HEX_PATTERN = Pattern.compile("&#([A-Fa-f0-9]{6})");
+    private static final Pattern LEGACY_TAG_PATTERN = Pattern.compile("[§&]([0-9a-fk-orxA-FK-ORX])");
+    private static final Pattern LEGACY_HEX_PATTERN = Pattern.compile("§x(§[0-9a-fA-F]){6}");
+
+    private static final MiniMessage MM = MiniMessage.miniMessage();
+
+    /**
+     * Translates RGB color codes (&#RRGGBB) into Minecraft color codes.
+     *
+     * @param message The message to translate.
+     * @return The translated message with RGB colors applied.
+     */
+    public static @NotNull String translateRGBColors(@Nullable String message) {
+        if (message == null || message.isEmpty()) {
+            return "";
+        }
+
+        Matcher matcher = HEX_PATTERN.matcher(message);
+
+        StringBuffer translatedMessage = new StringBuffer();
+
+        while (matcher.find()) {
+            String hexColor = matcher.group(1); // Extract the RRGGBB part
+            String minecraftColor = ChatColor.of("#" + hexColor).toString(); // Convert to Minecraft color
+            matcher.appendReplacement(translatedMessage, minecraftColor);
+        }
+
+        matcher.appendTail(translatedMessage);
+        return translatedMessage.toString();
+    }
+
+    public static @NotNull String legacyTagToMinimessage(@Nullable ChatColor color) {
+        if (color == null) {
+            return "";
+        }
+        switch (color.getName().toLowerCase()) {
```

Comment:
Could this not just be return `"<" + color.getName().toLowerCase() + ">"` as you have covered all cases?